### PR TITLE
[WIP] Allow using $bc.yml instead of crowbar.yml for barclamps

### PR DESCRIPTION
**Pull request by vuntz**
As we want to merge barclamps in one common git repo, we need to rename
crowbar.yml files to files with unique names, hence $bc.yml.

### CODE REVIEW

File: scripts/barclamp_create.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@
     path = ARGV[2] || "#{barclamp_dir}"
     target = File.join path, bc
     
-    if File.exist? File.join target, "crowbar.yml"
+    if !get_crowbar_yml_path(target).nil?
```
Comment (tboerger) [inline scripts/barclamp_create.rb]: I would prefer ```unless get_crowbar_yml_path(target).nil?```

File: scripts/barclamp_install.rb [hunk 2]
```diff
@@ -75,15 +75,15 @@ def rm_tmpdir(tmpdir)
     system "tar xzf \"#{src}\" -C \"#{tmpdir}\""
     # get the barclamp name, allow the tar ball to be named <barclampname>-barclamp.t*
     target="#{tmpdir}/#{src.split("/")[-1].split(".")[0].chomp("-barclamp")}"
-    if File.exists?(File.join(target,"crowbar.yml"))
+    if !get_crowbar_yml_path(target).nil?
```
Comment (tboerger) [inline scripts/barclamp_install.rb]: I would prefer ```unless get_crowbar_yml_path(target).nil?```

File: scripts/barclamp_install.rb [hunk 3]
```diff
@@ -75,15 +75,15 @@ def rm_tmpdir(tmpdir)
...
       candidates << target
     else
       puts "#{src} is not a barclamp tarball, ignoring."
     end
-  when File.exists?(File.join(src,"crowbar.yml"))
+  when !get_crowbar_yml_path(src).nil?
```
Comment (tboerger) [inline scripts/barclamp_install.rb]: I would prefer ```unless get_crowbar_yml_path(target).nil?```

File: scripts/barclamp_install.rb [hunk 4]
```diff
@@ -75,15 +75,15 @@ def rm_tmpdir(tmpdir)
...
     # We were handed something that looks like a path to a barclamp
     candidates << File.expand_path(src)
-  when File.exists?(File.join(BARCLAMP_PATH, src,"crowbar.yml"))
+  when !get_crowbar_yml_path(File.join(BARCLAMP_PATH, src))
```
Comment (tboerger) [inline scripts/barclamp_install.rb]: I would prefer ```unless get_crowbar_yml_path(File.join(BARCLAMP_PATH, src)).nil?```

File: scripts/barclamp_install.rb [hunk 5]
```diff
@@ -134,8 +134,8 @@ def rm_tmpdir(tmpdir)
       target=File.join(BARCLAMP_PATH, bc[:src].split("/")[-1])
       if File.directory? target
         debug "target directory #{target} exists"
-        if File.exists? "#{target}/crowbar.yml"
-          debug "#{target}/crowbar.yml file exists"
+        if !get_crowbar_yml_path(target).nil?
```
Comment (tboerger) [inline scripts/barclamp_install.rb]: I would prefer ```unless get_crowbar_yml_path(target).nil?```

File: scripts/barclamp_mgmt_lib.rb [hunk 6]
```diff
@@ -67,6 +67,20 @@ def fatal(msg, log = nil, exit_code = 1)
...
+    suggested_bc_name = directory.split(File::SEPARATOR)[-1]
+  end
+
+  # Look for both old name (crowbar.yml) and new name ($barclamp.yml)
+  ["#{suggested_by_name}.yml", "crowbar.yml"].each do |basename|
+    path = File.join(directory, basename)
+    return path if File.exists?(path)
+  end
+
+  nil
```
Comment (tboerger) [inline scripts/barclamp_mgmt_lib.rb]: If I see that correctly no need for that as it would be nil anyway.

Comment (vuntz) [inline scripts/barclamp_mgmt_lib.rb]: You're correct, but I prefer to be explicit about it to make it clear that it's intended. The alternative is to put a comment, but then it's the same amount of lines :-)

File: scripts/barclamp_install.rb (hunk 4 above)
Comment (vuntz) [inline scripts/barclamp_install.rb]: Hrm, that's a `when`, not a `if` :-)

File: scripts/barclamp_mgmt_lib.rb (hunk 6 above)
Comment (tboerger) [inline scripts/barclamp_mgmt_lib.rb]: That will be complained by rubocop 8)

File: scripts/barclamp_install.rb (hunk 4 above)
Comment (tboerger) [inline scripts/barclamp_install.rb]: Sorry, too much comments in the rush :D

**Comment by tboerger** [comment]
:+1:

**Comment by rsalevsky** [comment]
:+1:

**Pull request merged**